lib/rest-api.php

```diff
@@ -265,6 +265,44 @@ function gutenberg_add_block_format_to_post_content( $response, $post, $request
 	return $response;
 }
 
+/**
+ * Include target schema attributes to links, based on whether the user can.
+ *
+ * @param WP_REST_Response $response WP REST API response of a post.
+ * @param WP_Post          $post The post being returned.
+ * @param WP_REST_Request  $request WP REST API request.
+ * @return WP_REST_Response Response containing the new links.
+ */
+function gutenberg_add_target_schema_to_links( $response, $post, $request ) {
+	$new_links  = array();
+	$orig_links = $response->get_links();
+	$post_type  = get_post_type_object( $post->post_type );
+	// Only Posts can be sticky.
+	if ( 'post' === $post->post_type ) {
+		if ( current_user_can( $post_type->cap->edit_others_posts )
+			&& current_user_can( $post_type->cap->publish_posts ) ) {
```

Comment:
> Why is the capability for publishing posts required to make a post sticky?

I'm using the check within `edit_post()` as my point of reference: https://github.com/WordPress/WordPress/blob/3266b10d04ad43b9d983e0b55d4d6247be33e18a/wp-admin/includes/post.php#L401

> More importantly though, this should use the meta capabilities for the specified post and rely on `map_meta_cap()` instead of handling the logic manually.

I'm not sure I follow. Can you explain why?

Reply:
> Therefore I think the check in `edit_post()` is not accurate either. For this very case, there's no appropriate capability yet, so I'd recommend to introduce `stick_post` and `unstick_post` meta capabilities and add a clause for them in `map_meta_cap()`. We can then resolve it to the primitive capabilities we want to in there (if those are decided to be what currently happens in `edit_post()`, that's alright I guess). The benefit is that we then don't have to memorize all over what to check for when making a post sticky or unsticky, we can simply use intuitive `stick_post` / `unstick_post`, both in the current backend, the REST API, anywhere.

Does what you describe need to be solved for in this pull request?